plotly/tools.py

```diff
@@ -352,35 +352,36 @@ def embed(file_owner_or_url, file_id=None, width="100%", height=525):
     file_id (default=None) -- an int or string that can be converted to int
                               if you're using a url, don't fill this in!
     width (default="100%") -- an int or string corresp. to width of the figure
-    height (default="525") -- same as width but corresp. to the height of the figure
+    height (default="525") -- same as width but corresp. to the height of the
+                              figure
 
     """
     try:
-        s = get_embed(file_owner_or_url, file_id, width, height)
+        s = get_embed(file_owner_or_url, file_id=file_id, width=width,
+                      height=height)
+
         # see if we are in the SageMath Cloud
         from sage_salvus import html
         return html(s, hide=False)
     except:
         pass
     if _ipython_imported:
-        if file_id:
-            plotly_domain = (
-                session.get_session_config().get('plotly_domain') or
-                get_config_file()['plotly_domain']
-            )
-            url = "{plotly_domain}/~{un}/{fid}".format(
-                plotly_domain=plotly_domain,
-                un=file_owner_or_url,
-                fid=file_id)
-        else:
-            url = file_owner_or_url
+        url = file_owner_or_url
         return PlotlyDisplay(url, width, height)
     else:
-        warnings.warn(
-            "Looks like you're not using IPython or Sage to embed this plot. "
-            "If you just want the *embed code*, try using `get_embed()` "
-            "instead."
-            "\nQuestions? support@plot.ly")
+        if (session.get_session_config()['plotly_domain'] !=
+                'https://plot.ly'):
+            warnings.warn(
+                "Looks like you're not using IPython or Sage to embed this "
+                "plot. If you just want the *embed code*, try using "
+                "`get_embed()` instead."
+                "\nQuestions? support@plot.ly")
+        else:
+            warnings.warn(
+                "Looks like you're not using IPython or Sage to embed this "
+                "plot. If you just want the *embed code*, try using "
+                "`get_embed()` instead."
+                "\nQuestions? feedback@plot.ly")
```

Comment:
:cow2: keep it DRY \o/

Duplications here:

* 90% of that string
* `https://plot.ly` is defined as a default elsewhere

Here's a PR to help out with the `https://plot.ly` checking: #272 feel free to merge that in whenever, I believe we'll need it if we want to return these optional feedback emails.

If that's merged, I'd write this as folows:

```
if get_config_defaults()['plotly_domain'] != get_session_config()['plotly_domain']:
    feedback_email = 'feedback@plot.ly'
else:
    feedback_email = 'support@plot.ly'  # different domain likely means enterprise
warnings.warn('...\nQuestions? {}'.format(feedback_email))
```

And if you *really* want to be DRY, you could also centralize the feedback emails! (but i'd suggest another PR for that) I'll create an issue for it \o/